CHANGELOG.md

```diff
@@ -23,11 +25,24 @@
 - `az_iot_provisioning_client_init()` parameter `global_device_endpoint` renamed to `global_device_hostname`.
 - Renamed the macro `AZ_SPAN_NULL` to `AZ_SPAN_EMPTY`.
 - Renamed the `az_result` value `AZ_ERROR_INSUFFICIENT_SPAN_SIZE` to `AZ_ERROR_NOT_ENOUGH_SPACE`.
+- Made the helper macros `AZ_RETURN_IF_FAILED` and `AZ_RETURN_IF_NOT_ENOUGH_SIZE` internal.
```

Comment:
We probably should say that we removed it rather than made internal - If you read Changelog from the customer perspective, AZ_RETURN... it is no longer available, it is removed from public header. Plus it no longer has public "AZ_" prefix.. So to a large extent made internal is indistinguishable from being deleted, as it should be.

Reply:
I think I agree with Anton here. From a customer perspective, we would rather express to them the ultimate usability. If we say we made it internal, it exposes internal implementation and muddies their mental model of what is available to them. If we just say "we removed it", they know they have no access to it anymore.